pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -609,8 +609,11 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	k8sutil.SetOwnerRef(&deployment.ObjectMeta, &c.clusterInfo.OwnerRef)
+	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if !osdProps.onPVC() {
-		cephv1.GetOSDPlacement(c.spec.Placement).ApplyToPodSpec(&deployment.Spec.Template.Spec)
+		p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
```

Comment:
In addition to the OSD daemons, we also need to apply it to the osd prepare pods.

Reply:
Done.